content/blog/2022/2022-09-25-node-urlShortener.md

```diff
@@ -0,0 +1,494 @@
+- shortid: Generates non-sequential short unique ids
+
+Next, create an `index.js` and a  `.env` file.
+
+Our application should be structured like this:
+{{% image alt="We want to deploy our application into a staging and a production environment" src="images/posts/designing-a-aws-cdk-project/serverapp.png" %}}
+
+In the `index.js` file, paste this to create a simple`Node.Js` server:
+```javascript
+const express = require('express');
+const app = express();
+
+// Server Setup
+const PORT = process.env.PORT || 3333;
+app.listen(PORT, () => {
+  console.log(`Server is running at PORT: ${PORT}`);
+});
+```
+In the code above, we created an express server by importing and instantiating the express package into the index.js file. Making it listen on our custom PORT `3333`.
+
+Then next in our `.env` file we are going to store our `DOMAIN_URL` and `MongoDB_URI`. For the time being, the `DOMAIN_URL` in this application will be our `localhost` server location. 
+
+To get our `MongoDB_URI` we have to create a MongoDB atlas cluster
+
+## Creating MongoDB Atlas Cluster
+Follow these steps to create a MongoDB altas cluster:
+- [Go here](https://account.mongodb.com/account/register)  to sign up for a new MongoDB Atlas account.
+- Fill in the registration form with your information and click  **Sign up**.
+- Click on Deploy a shared cloud database for `Free`
+- Click on Create a Shared Cluster
+- Next, click on **Database access** on the sidebar and Add New Database User
+- Select Password then enter in a username and password detail for your user.
+- `Built-in Role` select `Atlas Admin`
+- Click the **Add User** button to create your new user
+- Next, click on **Network Access** on the sidebar.
+To Allow Access From All IP Addresses
+- Click on Add IP Address button
+- Select `ALLOW ACCESS FROM ANYWHERE`
+- Click the **Confirm**  button.
+- Next, click on **Database** on the sidebar
+- Click the  **Connect**  button for your cluster
+- In the popup modal, click on  **Connect your application**.
+- **Copy** the URI on the clipboard
+- Lastly, all you need to do is replace the `<password>` field with the password you created previously.
+
+Our MongoDB atlas is all set and ready for use. Otherwise, click [here](https://www.freecodecamp.org/news/get-started-with-mongodb-atlas/) for a more in-depth guide on how to set up a MongoDB cluster.
+
+We can now, store our `DOMAIN_URL` and `MongoDB_URI` in `.env`:
+```
+MONGO_URI= mongodb+srv://<username>:<password>@cluster0.oq1hdin.mongodb.net/?retryWrites=true&w=majority
+DOMAIN_URL=http://localhost:3333
+```
+
+Enter the following code into the terminal to Start our Node.js server:
+```
+node index.js
+```
+Output:
+```
+Server is running at PORT: 3333
+```
+## Creating MongoDB Schema
+Everything in Mongoose starts with a Schema. Each schema maps to a MongoDB collection and defines the shape of the documents within that collection. We will need to create a Schema for our Urls. 
```

Comment:
What is Mongoose? I know you listed it above in the dependencies, but maybe add a sentence here to remind the reader what it is.